File: test6/equi_filter.pl
#!/usr/bin/perl -w


@array = <STDIN>;
chomp @array;
foreach $IN (@array){
	@line = split(/ /, $IN);
	foreach $word (@line){
		@temp = split(//, $word);
		foreach $sub (@temp){
			if (exists($count_hash{lc($sub)})){
				$count_hash{lc($sub)} += 1;
			}
			else{
				$count_hash{lc($sub)} = 1;
			}
		}
		$count = $count_hash{lc($temp[0])};
		foreach $sub (@temp){
			if ($count_hash{lc($sub)} != $count){
				$word = 1;}}
		foreach $sub (@temp){
			$count_hash{lc($sub)} = 0;}
	}
	@line =grep(/[^1]/, @line);
	$A = join(" ", @line);
	push @new, $A;
}

foreach $d (@new){
	print "$d\n";}

File: test6/equi_filter.py
#!/usr/bin/python3
import sys


dic = {}
L=[]
for i in sys.stdin:
    line = i.split( )
    new = []
    for word in line:
        A=0
        for j in range(len(word)):
            dic[word[j].lower()] = 0
        for j in range(len(word)):
            dic[word[j].lower()] += 1
        count = dic[word[0].lower()]
        for j in range(len(word)):
            if count != dic[word[j].lower()]:
                A = 1
        for j in range(len(word)):
            dic[word[j].lower()] = 0
        if not A:
            new.append(word)
    lines = ' '.join(new)
    L.append(lines)
for a in L:
    print(a)



